routes/ScannerExchangePage.vue

```diff
@@ -0,0 +1,417 @@
+          style="max-width: 500px">
+          <div v-if="requestOrigin">
+            <chapi-header
+              :name="relyingPartyName"
+              :image="relyingPartyImage"
+              :request="relyingPartyRequest" />
+            <q-separator class="q-my-md" />
+          </div>
+          <problem-card
+            v-if="error"
+            :error="error"
+            :account="account"
+            :loading="loading"
+            @cancel="$event.waitUntil(cancel(error))" />
+          <share-credentials
+            v-else-if="!loading && display === 'share'"
+            :verifiable-presentation-request="verifiablePresentationRequest"
+            :request-origin="requestOrigin"
+            @cancel="$event.waitUntil(cancel())"
+            @share="$event.waitUntil(share($event.presentation))" />
+          <store-credentials
+            v-else-if="!loading && display === 'store'"
+            :account="account"
+            :holder="holder"
+            :verifiable-credential="verifiableCredential"
+            @cancel="$event.waitUntil(cancel())"
+            @store="$event.waitUntil(store($event))" />
+        </div>
+      </div>
+    </div>
+  </q-page>
+</template>
+
+<script>
+/*!
+ * Copyright (c) 2018-2024 Digital Bazaar, Inc. All rights reserved.
+ */
+
+import {computed, ref, toRaw} from 'vue';
+import {exchanges, getCredentialStore, helpers} from '@bedrock/web-wallet';
+import BarcodeScanner from '../components/BarcodeScanner.vue';
+import ChapiHeader from '../components/ChapiHeader.vue';
+import {Html5QrcodeSupportedFormats} from 'html5-qrcode';
+import {httpClient} from '@digitalbazaar/http-client';
+import ProblemCard from '../components/ProblemCard.vue';
+import {sessionDataRef} from '../lib/session.js';
+import ShareCredentials from '../components/ShareCredentials.vue';
+import StoreCredentials from '../components/StoreCredentials.vue';
+import {useQuasar} from 'quasar';
+import {useRouter} from 'vue-router';
+import {WebAppManifestClient} from '@digitalbazaar/web-app-manifest-utils';
+
+export default {
+  name: 'WalletScanner',
+  components: {
+    ChapiHeader,
+    ProblemCard,
+    BarcodeScanner,
+    ShareCredentials,
+    StoreCredentials
+  },
+  setup() {
+    // constants
+    const {prettify} = helpers;
+    const manifestClient = new WebAppManifestClient();
+    const supportedFormats = [Html5QrcodeSupportedFormats.QR_CODE];
+
+    // refs
+    const holder = ref('');
+    const ready = ref(null);
+    const error = ref(null);
+    const reject = ref(null);
+    const result = ref(null);
+    const exchange = ref(null);
+    const storing = ref(false);
+    const display = ref('login');
+    const exchanging = ref(false);
+    const registering = ref(false);
+    const enableScanning = ref(true);
+    const verifiableCredential = ref([]);
+    const verifiablePresentation = ref();
+    const credentialRequestOrigin = ref('');
+    const verifiablePresentationRequest = ref();
```

Comment:
We don't have to do this in this PR, but I do wonder if this is all cleaner and more easily done with this pattern:

```js
const state = reactive({
  holder: '',
  ...
});
```

And then just `state.holder` etc. throughout.

Reply:
It also seems like you can combine `reactive` and `computed` without issue:

https://stackoverflow.com/a/65977617

Reply:
I turned this into an issue so we can keep the code the same for now as that other page -- when we consolidate we should try to also clean this up.